wallet/wallet-cli-lib/tests/cli_test_framework.rs

```diff
@@ -320,27 +305,59 @@ impl CliTestFramework {
             vi_mode: false,
         };
 
-        let console = MockConsole::new(commands);
+        let (output_tx, output_rx) = std::sync::mpsc::channel();
+        let (input_tx, input_rx) = std::sync::mpsc::channel();
 
-        tokio::time::timeout(
-            Duration::from_secs(60),
-            wallet_cli_lib::run(
-                console.clone(),
-                wallet_options,
-                Some(Arc::clone(&self.chain_config)),
-            ),
-        )
-        .await
-        .unwrap()
-        .unwrap();
+        let input = MockConsoleInput { input_rx };
 
-        let res = console.output.lock().unwrap().clone();
-        res
+        let output = MockConsoleOutput { output_tx };
+
+        let wallet_task = tokio::spawn(async move {
+            tokio::time::timeout(
+                Duration::from_secs(120),
+                wallet_cli_lib::run(input, output, wallet_options, Some(chain_config)),
+            )
+            .await
+            .unwrap()
+            .unwrap();
+        });
```

Comment:
Just curious: Why did you decide to move this into a separate task? Is it because of the timeout?

Reply:
The tests now work differently, `MockConsoleInput` now sends input and `MockConsoleOutput` is expected output. For this to work, the CLI wallet task should be running in the background.